driven/storage/adapter.go

```diff
@@ -4305,6 +4305,9 @@ func (sa *Adapter) getFilterForParams(params map[string]interface{}) bson.M {
 			filter[k] = v
 		} else if v == "$exists" {
 			filter[k] = bson.M{"$exists": true}
+		} else if reflect.TypeOf(v).Kind() == reflect.String {
```

Comment:
Why do you check for String. Doesn't it come as slice? I see there is already a check for slice.